src/manager.hpp

```diff
@@ -200,13 +204,28 @@ class Manager
      *
      * @param[in] dataSyncCfg - The data sync config to sync
      * @param[in] srcPath - The modified path inside the cfg path, if available.
+     * @param[in] retryCount - The current retry attempt count
      *
      * @return Returns true if sync succeeds; otherwise, returns false
      *
      */
     sdbusplus::async::task<bool>
         syncData(const config::DataSyncConfig& dataSyncCfg,
-                 fs::path srcPath = fs::path{});
+                 fs::path srcPath = fs::path{}, size_t retryCount = 0);
+
+    /**
+     * @brief Retry the data sync operation based on failure
+     *
+     * @param[in] cfg - Data sync configuration
+     * @param[in] srcPath - Source path to be synced
+     * @param[in] retryCount - Current retry attempt number
+     * @param[in] rsyncExitCode - Exit code from the last rsync attempt
```

Comment:
why is it required? is it really worth? 

We will have error trace the sync is failed, right? 

Please avoid adding redundant traces.